### PR TITLE
Feature/resharper 2016.2

### PR DESCRIPTION
**Pull request by CodingGorilla**
Updates for Resharper 2016.2.

### CODE REVIEW

**Comment by CodingGorilla** [comment]
I also updated the reference to `Newtonsoft.JSON` as I know the 9.x branch had some pretty significant performance improvements helped along by the ASP.NET team.

**Comment by MrJul** [comment]
You can remove this commit: Json.NET isn't included with Enhanced Tooltip, the one bundled with ReSharper will be used so it's better to keep the old one to avoid problems such as using new methods from v9 which won't exist at runtime.

(EnhancedTooltip doesn't use Json.NET directly so it shouldn't matter, but let's keep it safe.)

File: GammaJul.ReSharper.EnhancedTooltip/VisualStudio/QuickInfoSourceBase.cs [hunk 1]
```diff
@@ -49,8 +49,9 @@ protected ITextBuffer TextBuffer
 			IDocument document = new JetVsTextBuffer(bufferAdapter).JetDocument.Value;
 			if (document == null)
 				return null;
-			
-			IDocumentMarkup documentMarkup = DocumentMarkupManagerBase.TryGetMarkupModel(document);
+
+
+		    IDocumentMarkup documentMarkup = Shell.Instance.GetComponent<IDocumentMarkupManager>().TryGetMarkupModel(document);
```
Comment (MrJul) [inline GammaJul.ReSharper.EnhancedTooltip/VisualStudio/QuickInfoSourceBase.cs]: Please use `Shell.Instance.TryGetComponent<IDocumentMarkupManager>()?.TryGetMarkupModel(document)` instead, I want `AugmentQuickInfoSession` do be safe (do nothing if we don't have everything we need) before calling `AugmentQuickInfoSessionCore`.

**Comment by MrJul** [comment]
I'm one of those evil guys that use tabs instead of spaces (don't judge!). Please fix the formatting :)

**Comment by CodingGorilla** [comment]
Will fix the tabs/spaces issue, regarding the `?.TryGetMarkupModel()`:

image

I can leave it if you like... :smile:

**Comment by MrJul** [comment]
Is it with `TryGetComponent`?

**Comment by CodingGorilla** [comment]
Yea:

image

**Comment by MrJul** [comment]
That's `GetComponent`, not `TryGetComponent`, unless I'm misunderstanding something :grinning:

**Comment by CodingGorilla** [comment]
Sorry, reading failure... changes incoming.

**Comment by MrJul** [comment]
Thank you for the PR! It's looking good. I'll test, and publish it on the gallery asap.

**Comment by CodingGorilla** [comment]
Glad to help!

**Pull request merged**